### DISCUSSION TITLE
installing packages that have no checksum

### DISCUSSION THREAD

**Discussion by 101Dude** [Homebrew/discussions#5206] [Everyday usage] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.2.12-59-ga3e5e3f
ORIGIN: https://github.com/Homebrew/brew
HEAD: a3e5e3f7a0e98fe8ffbd8894a9ea0c1fb5c9884f
Last commit: 25 hours ago
Core tap HEAD: 6ec4be845d0533db9d898ede4745b3edfcb0ac02
Core tap last commit: 3 weeks ago
Core tap JSON: 15 Mar 19:45 UTC
Core cask tap HEAD: a809c91bf6a78ffd31d886caf784df04d6a3e80a
Core cask tap last commit: 3 weeks ago
Core cask tap JSON: 15 Mar 19:45 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: ["--greedy", "--require-sha", "--no-quarantine", "--appdir=/Applications", "--fontdir=/Library/Fonts", "--qlplugindir=/Library/QuickLook", "--internet_plugindir=/Library/Internet Plug-Ins", "--prefpanedir=/Library/PreferencePanes/", "--servicedir=/Library/Services"]
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.6u9OZEdpNf/org.xquartz:0
HOMEBREW_EDITOR: nano
HOMEBREW_GITHUB_API_TOKEN: set
HOMEBREW_MAKE_JOBS: 16
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 3.1.4 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: 16-core 64-bit kabylake
Clang: 15.0.0 build 1500
Git: 2.43.2 => /usr/local/bin/git
Curl: 8.4.0 => /usr/bin/curl
macOS: 14.4-x86_64
CLT: 15.3.0.0.1.1708646388
Xcode: 15.3
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed kegs have no formulae!
This means they were either deleted or installed manually.
You should find replacements for the following formulae:
  python-platformdirs
  python-distlib
  python-filelock
  python-mutagen
  python-markupsafe
```

### Description of issue

I tried to install chromium and this is what I see.

```
Warning: Cannot verify integrity of '1315ff71c24df00ce56f202a5fb9e9fe523da78f709f32480ed3fa2e171b31eb--chromium.rb'.
No checksum was provided.
For your reference, the checksum is:
  sha256 "e8935c82efd55f299f6988e52b1fb739b876914ee5ae749104abd7972d759d38"
Error: Cask 'chromium' does not have a sha256 checksum defined and was not installed.
This means you have the --require-sha option set, perhaps in your HOMEBREW_CASK_OPTS.
````

I don't understand the prompts here. It says no checksum was provided and then it goes on to tell me the checksum.

Does this mean it was calculated after the fact?
If so the prompt can communicate that instead. Something like:

> For your reference, we calculated the checksum:

Second. why not add an additional prompt that suggests using `--force` to bypass so users don't run to edit their dot files each time they see this prompt?

Thanks you in advance for good faith responses.

**Comment by SMillerDev** [comment]
> Does this mean it was calculated after the fact?
> If so the prompt can communicate that instead. Something like:

That sounds good, could you open a pull request for that?

> Second. why not add an additional prompt that suggests using --force to bypass so users don't run to edit their dot files each time they see this prompt?

Because people who add default cask options should know what they set and how to resolve issues that arise from that.

**Reply by 101Dude** [reply]
> > Second. why not add an additional prompt that suggests using --force to bypass so users don't run to edit their dot files each time they see this prompt?
> 
> Because people who add default cask options should know what they set and how to resolve issues that arise from that.

People should know a lot of things... first is that people tend to set and forget. A complicating factor is that this is not mentioned in the `brew install --help` section for `--force`

>   -f, --force                      Install formulae without checking for
>                                    previously installed keg-only or non-migrated
>                                    versions. When installing casks, overwrite
>                                    existing files (binaries and symlinks are
>                                    excluded, unless originally from the same
>                                    cask).
> 

So "knowing how" comes down to a web search and believing 3rd parties?

**Comment by Bo98** [comment, marked as answer]
The specific situation you had was a bug. The checksum you were seeing was for the `.rb` file rather than the `chromium` download itself and that message should not have been displayed for that. This has been fixed now.

**Reply by 101Dude** [reply]
Thank you!

I still feel that adding a note about using `--force` in the prompt is the best place for this situation. Information at hand when it's required is helpful.
